grizzly/users/restapi.py

```diff
@@ -603,22 +630,32 @@ def _request(self, client: Union[FastHttpSession, ResponseEventSession], request
         name = f'{request.scenario.identifier} {request_name}'
 
         if payload is not None:
-            try:
-                parameters['json'] = json.loads(payload)
-            except json.decoder.JSONDecodeError as exception:
-                # so that locust treats it as a failure
-                self.environment.events.request.fire(
-                    request_type=RequestType.from_method(request.method),
-                    name=name,
-                    response_time=0,
-                    response_length=0,
-                    context=self._context,
-                    exception=exception,
-                )
-                logger.error(f'{url}: failed to decode: {payload=}')
-
-                # this is a fundemental error, so we'll always stop the user
-                raise StopUser()
+            if request.response.content_type == TransformerContentType.JSON:
+                try:
+                    parameters['json'] = json.loads(payload)
+                except json.decoder.JSONDecodeError as exception:
+                    # so that locust treats it as a failure
+                    self.environment.events.request.fire(
+                        request_type=RequestType.from_method(request.method),
+                        name=name,
+                        response_time=0,
+                        response_length=0,
+                        context=self._context,
+                        exception=exception,
+                    )
+                    logger.error(f'{url}: failed to decode: {payload=}')
+
+                    # this is a fundemental error, so we'll always stop the user
+                    raise StopUser()
+            elif request.response.content_type == TransformerContentType.MULTIPART_FORM_DATA:
+                if arguments is None or 'multipart_form_data_name' not in arguments or 'multipart_form_data_filename' not in arguments:
```

Comment:
is it possible to have this check when creating the request? then it would fail before the load test scenario has started, as in now it will fail when locust execute this request.

Reply:
check!